exercises/tom_clayton/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Import statements
+ */
+
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <algorithm>
+#include <cctype>
+#include <string.h>
+#include <vector>
+#include <map>
+using namespace std;
+
+/**
+ * Function to return a string filepath from user input
+ */
+
+std::string getInputFn() {
+  std::cout << "Please enter the path of the file to be analysed" << std::endl;
+  std::string fin;
+  std::cin >> fin;
+  return fin;
+}
+
+/**
+ * Function to split up a string and return a vector of words with
+ * len <4, and converted to lower case
+ */
+
+std::vector<std::string> splitByDelim(const std::string& s, char delim) {
+  
+  std::vector<std::string> tokens;
+  std::string token;
+  std::istringstream tokenStream(s);
+  
+  while (std::getline(tokenStream, token, delim)) {
+    if(token.size() < 4) {
+      continue;
+    }
+
+    std::transform(token.begin(), token.end(), token.begin(),
+    [](unsigned char c){ return std::tolower(c);});
+    
+    tokens.push_back(token);
+    //cout << token << endl;
+  }
+
+  return tokens;
+}
+
+/**
+ * Function to replace puncuation with whitespace in a string
+ */
+
+void depunctuate(std::string& line) {
+
+  std::replace_if(line.begin() , line.end(), 
+  [] (const char& c) { return std::ispunct(c) ;},' ');
```

Comment:
I'd not come across ispunct before.  
The list of characters that ispunct considers to be punctuation seems to be longer than the list specified in the exercise so please could you change this to use the shorter list in the exercise: .,?'"!():
Also the behaviour here for apostrophes isn't quite what I was expecting eg Engineer's should be converted to Engineers however this code ends up counting it as Engineer. Maybe the wording of the exercise isn't 100% clear but that's what the intention was...
Regardless of that - the approach of converting hyphens to spaces is nice so that you can just identifying words later on by using a space delimiter.

Reply:
Replaced this with a new function that determines whether a given character is to be removed, replaced with whitespace or left. specPunct(char c) in new commit